# 04 One-sample and Two-sample test

## 01 One-sample test

### 1. One-sample t test

- <img src="https://latex.codecogs.com/svg.image?(x_1,\cdots,x_n)\sim&space;N(\mu,\sigma^2)"><br>
<img src="https://latex.codecogs.com/svg.image?H_0:\mu=\mu_0\;\;vs\;\;H_1:\mu\neq\mu_0">

- 검정통계량 <img src="https://latex.codecogs.com/svg.image?t_0=\frac{\bar{x}-\mu_0}{s/\sqrt{n}}\sim&space;t(n-1)">

ex) example concerning daily intake in kJ for 11 women(Altman, 1991, p.183). To investigate whether the women's energy intake deviates systematically from a recommended value of 7725kJ.<br>

예) 11명의 여성들의 일일 에너지 섭취량에 대한 예제를 살펴보겠습니다(Altman, 1991, p.183). 이 여성들의 에너지 섭취량이 권장값인 7725kJ에서 체계적으로 벗어나는지 조사하기 위해서입니다.


In [2]:
daily.intake <- c(5260, 5470, 5640, 6180, 6390, 6515, 6805, 7515, 7515, 8230, 8770)
mean(daily.intake)
sd(daily.intake)
quantile(daily.intake)

[1] 6753.636

[1] 1142.123

0%  25%  50%  75% 100% 
5260 5910 6515 7515 8770

In [3]:
t.test(daily.intake, mu=7725)


	One Sample t-test

data:  daily.intake
t = -2.8208, df = 10, p-value = 0.01814
alternative hypothesis: true mean is not equal to 7725
95 percent confidence interval:
 5986.348 7520.925
sample estimates:
mean of x 
 6753.636 


- You can immediately see that p<0.05 and thus that (using the customary 5% level of significance) data deviate significantly from the hypotheses that the mean is 7725.
- Optional arguments
  - mu (def=0)
  - alternative = "grater"("g") or "less"("l") (def=two-sided)
  - conf.level = 0.99 (def=0.95)

### 2. Wilcoxon signed-rank test

- The t tests are fairly robust against departures from the normal distribution especially in larger samples, but sometimes you wish to avoid making that sssumption. To this end, the distribution-free methods are convenient.
- t 검정은 특히 큰 샘플에서는 정규 분포에서 벗어난 경우에도 상당히 견고합니다. 하지만 때로는 그 가정을 피하고 싶을 수 있습니다. 이를 위해 비모수적인 방법들이 편리합니다.

- For the one-sample Wilcoxon test, the procedure is to subtract is to subtract the theoretical <img src="https://latex.codecogs.com/svg.image?\mu_0"> and rank the differences according to their numerical value, ignoring the sign, and then calculate the sum of the positive or negative ranks.
- 단일표본 Wilcoxon 검정에서의 절차는 이론적인 <img src="https://latex.codecogs.com/svg.image?\mu_0">를 빼고 그 차이를 숫자 값에 따라 순위를 매기되 부호를 무시하고, 그 다음 양수 또는 음수 순위의 합계를 계산하는 것입니다.

In [4]:
wilcox.test(daily.intake, mu=7725)

Warning message in wilcox.test.default(daily.intake, mu = 7725):
"tie가 있어 정확한 p값을 계산할 수 없습니다"



	Wilcoxon signed rank test with continuity correction

data:  daily.intake
V = 8, p-value = 0.0293
alternative hypothesis: true location is not equal to 7725


- The test statistic V is the sum of the positive ranks. In the example, the p-value is computed from the normal approximation because of the tie at 7515.
- 검정 통계량 V는 양수 순위의 합계입니다. 예제에서 7515에서의 동률로 인해 정규 근사를 사용하여 p값이 계산됩니다.

## 02 Two-sample test(Independent samples)

### 1. Two-sample test

- The two-sample t test is used to test the hypothesis that two samples may be assumed to come from distributions with the same mean <img src="https://latex.codecogs.com/svg.image?(\delta_0=0)">.
- 두 표본 t 검정은 두 표본이 동일한 평균 <img src="https://latex.codecogs.com/svg.image?(\delta_0=0)">을 가진 분포에서 추출되었을 것으로 가정하는 가설을 검정하는 데 사용됩니다.
- Hypothesis for comparing two population means <img src="https://latex.codecogs.com/svg.image?(\mu_1,\mu_2)">.

- ① <img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"> <br>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2>\delta_0">
- ② <img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"> <br>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2<\delta_0">
- ③ <img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"> <br>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2\neq\delta_0">

- If the sample size is sufficiently large (usually greater than 30), the following test statistic is used regardless of the population distribution(standard normal distribution).
- 표본 크기가 충분히 큰 경우 (일반적으로 30 이상), 모집단 분포에 관계없이 다음 검정 통계량이 사용됩니다(표준 정규분포).
  - <img src="https://latex.codecogs.com/svg.image?Z=\frac{(\bar{X_1}-\bar{X_2})-\delta_0}{\sqrt{\frac{S_1^2}{n_1}&plus;\frac{S_2^2}{n_2}}}">    

- If the sample size is small, under the assumption of normal distribution and equal variance we can use the following test statistics.
- 표본 크기가 작은 경우, 정규 분포를 가정하고 분산이 동일하다고 가정할 때 다음과 같은 검정 통계량을 사용할 수 있습니다.


|<img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"><br><img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2>\delta_0">|<img src="https://latex.codecogs.com/svg.image?\frac{(\bar{X_1}-\bar{X_2})-\delta_0}{\sqrt{\frac{S_p^2}{n_1}&plus;\frac{S_p^2}{n_2}}}>t_{n_1+n_2-2,\alpha} \Rightarrow reject\;H_0">|
|:-:|:-:|
|<img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"><br><img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2<\delta_0">|<img src="https://latex.codecogs.com/svg.image?\frac{(\bar{X_1}-\bar{X_2})-\delta_0}{\sqrt{\frac{S_p^2}{n_1}&plus;\frac{S_p^2}{n_2}}}<-t_{n_1+n_2-2,\alpha} \Rightarrow reject\;H_0">|
|<img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"><br><img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2\neq\delta_0">|<img src="https://latex.codecogs.com/svg.image?\left\|\frac{(\bar{X_1}-\bar{X_2})-\delta_0}{\sqrt{\frac{S_p^2}{n_1}&plus;\frac{S_p^2}{n_2}}}\right\|>t_{n_1&plus;n_2-2,\alpha/2}\Rightarrow&space;reject\;H_0">|

<br>
<img src="https://latex.codecogs.com/svg.image?S_p^2=\frac{(n_1-1)S_1^2&plus;(n_2-1)S_2^2}{n_1&plus;n_2-2}">

- If variances are not equal, we need to modify the degrees of freedom of the t distribution to *Φ*(Satterthwaite approximation)
- 분산이 동일하지 않은 경우, t 분포의 자유도를 Satterthwaite 근사치인 *Φ*로 수정해야 합니다.

|<img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"><br><img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2>\delta_0">|<img src="https://latex.codecogs.com/svg.image?\frac{(\bar{X_1}-\bar{X_2})-\delta_0}{\sqrt{\frac{S_1^2}{n_1}&plus;\frac{S_2^2}{n_2}}}>t_{\phi,\alpha} \Rightarrow reject\;H_0">|
|:-:|:-:|
|<img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"><br><img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2<\delta_0">|<img src="https://latex.codecogs.com/svg.image?\frac{(\bar{X_1}-\bar{X_2})-\delta_0}{\sqrt{\frac{S_1^2}{n_1}&plus;\frac{S_2^2}{n_2}}}<-t_{\phi,\alpha} \Rightarrow reject\;H_0">|
|<img src="https://latex.codecogs.com/svg.image?H_0:\mu_1-\mu_2=\delta_0"><br><img src="https://latex.codecogs.com/svg.image?H_1:\mu_1-\mu_2\neq\delta_0">|<img src="https://latex.codecogs.com/svg.image?\left\|\frac{(\bar{X_1}-\bar{X_2})-\delta_0}{\sqrt{\frac{S_1^2}{n_1}&plus;\frac{S_2^2}{n_2}}}\right\|>t_{\phi,\alpha/2}\Rightarrow&space;reject\;H_0">|

<br>
<img src="https://latex.codecogs.com/svg.image?\phi=\frac{\left[\frac{S_1^2}{n_1}&plus;\frac{S_2^2}{n_2}\right]^2}{\frac{(S_1^2/n_1)^2}{n_1-1}&plus;\frac{(S_2^2/n_2)^2}{n_2-1}}">





- The two-sample t test is used to test the hypothesis that two samples may be assumed to come from distributions with the same mean.
- 두 표본 t 검정은 두 표본이 동일한 평균을 가진 분포에서 추출되었을 것으로 가정하는 가설을 검정하는 데 사용됩니다.

In [4]:
library(ISwR)
data(energy)
attach(energy)

t.test(expend~stature)

The following objects are masked from energy (pos = 3):

    expend, stature




[1]  9.21  7.53  7.48  8.08  8.09 10.15  8.40 10.88  6.13  7.90 11.51 12.79
[13]  7.05 11.85  9.97  7.48  8.79  9.69  9.68  7.58  9.19  8.11


	Welch Two Sample t-test

data:  expend by stature
t = -3.8555, df = 15.919, p-value = 0.001411
alternative hypothesis: true difference in means between group lean and group obese is not equal to 0
95 percent confidence interval:
 -3.459167 -1.004081
sample estimates:
 mean in group lean mean in group obese 
           8.066154           10.297778 


- The confidence interval is for the difference in means and does not contain 0, which is in accordance with the p-value indicating a significant difference at the 5% level.
- 신뢰 구간은 평균 차이에 대한 것으로, 0을 포함하지 않으며, 이는 5% 수준에서 유의한 차이를 나타내는 p값과 일치합니다.

- It is Welch's variant of the t test. This is the test where you do not assume that the variance is the same in the two groups, which(among other things) results in the fractional degrees of freedom.
- 이는 웰치의 t 검정의 변형입니다. 이는 두 그룹의 분산이 동일하다고 가정하지 않는 테스트로, 이로 인해 분수 자유도가 발생합니다.


- To get the usual(textbook) t test, you must specify that you are willing to assume that the variances are the same. This is done via the optional argument var.equal=T;
- 표준(교과서) t 검정을 얻으려면 두 분산이 동일하다고 가정할 것을 명시해야 합니다. 이는 선택적 인자 var.equal=T를 통해 수행됩니다.

In [3]:
t.test(expend~stature, var.equal=T)


	Two Sample t-test

data:  expend by stature
t = -3.9456, df = 20, p-value = 0.000799
alternative hypothesis: true difference in means between group lean and group obese is not equal to 0
95 percent confidence interval:
 -3.411451 -1.051796
sample estimates:
 mean in group lean mean in group obese 
           8.066154           10.297778 


- Notice that the degrees of freedom now has become a whole number, namely 13+9-2=20. The p-value has dropped slightly (from 0.14% to 0.08%) and the confidence interval is a litter narrower, but overall the changes are slight.
- 주의해야 할 점은 자유도가 이제 정수가 된 것입니다. 즉, 13+9-2=20입니다. p값은 약간 감소했습니다 (0.14%에서 0.08%로), 신뢰 구간은 조금 더 좁아졌지만, 전반적으로 변화는 미미합니다.